solution-base/build.sh

```diff
@@ -194,7 +199,11 @@ function render_mongodb_sharded_yamls()
         --set 'configsvr.persistence.selector.matchLabels.app\.kubernetes\.io/name=mongodb-sharded-config' \
         --set 'configsvr.persistence.selector.matchLabels.app\.kubernetes\.io/part-of=zenko' \
         --set "configsvr.mongodbExtraFlags={--setParameter rollbackTimeLimitSecs=259200}" \
-        --set "shardsvr.dataNode.mongodbExtraFlags={--setParameter rollbackTimeLimitSecs=259200}" \
+        --set mongos.resources.requests.memory=${MONGODB_MONGOS_RAM_REQUEST} \
+        --set mongos.resources.limits.memory=${MONGODB_MONGOS_RAM_LIMIT} \
```

Comment:
The issue I have with patching the CR is that we introduce risks of having changes in the bitnami generated yaml files leading to incompatibilities, while having such variables is something we do already, and also let us test such configuration in Zenko CI, not only Artesca, as I believe MongoDB is Zenko-specific. For example, having multiple shards or more MongoDB services (e.g., arbitrer), will require deep changes in the "filling" logic of the Artesca Installer, while template variables are easy to replace and control...

However I agree with the fact that we might end up with many variables (although I am not sure if we need more as of today)

Do you have any preference, or is it a strong request for change?

Reply:
> Do you have any preference, or is it a strong request for change?

I think i may have a slight preference for patching the CR, just sharing my doubts here for your consideration.

IMHO patching the resource limits would be fine: the dependency is more on k8s than on the details of the chart... and we introduce more resources (artbiters...) we will need also corresponding patch in artesca installer to set the values, so we don't actually reduce the coupling, but just reduce the freedom...

That said, this is indeed not a request for change, and we probably don't need more variables _today_ (the point is more about maintenance in the future than fixing today, as is generally the case with engineering).

> About the "no value" thing, indeed it is not possible directly (without additional logic in the installer) with this approach, but the whole point is to enforce this change everywhere and for everyone.

That may be more concerning. While the "max memory limit" question has been debated, there may be other question/impacts of memory request, where some tweaks may be needed... And i would really want to avoid having to change zenko again just because we need to set a different value or remove one of the limits...

Can you try with an empty string (`""`) or 0, see if that allows reverting to the "unlimited" behavior while specifying the field?